### PR TITLE
sklearnserver: ignore converting instances into np.array

### LINKED ISSUE #1912: Sklearnserver - move numpy array convert to transformer
The sklearnserver perform [numpy array convert](https://github.com/kserve/kserve/blob/45c3c2cf187b5255eb918fa365606f20d0e906d2/python/sklearnserver/sklearnserver/model.py#L47) to the inputs. That means all array will have the same type of items. If a model receive different types, a custom server should be written.

I suggest to remove the convert, and if needed, it should be done in transformer preprocess.

### PR DESCRIPTION
**Pull request by Suresh-Nakkeran**
To support mix types of input such as pandas data frames, we remove the numpy array conversion in sklearnserver which can be done in transformer if needed.

fixes #1912

### CODE REVIEW

**Comment by yuzisun** [comment]
@Suresh-Nakkeran Can you help test with the mixed type [example](https://github.com/kserve/kserve/issues/1799) with your change?

File: python/sklearnserver/sklearnserver/sklearn-multi-datatype-model.ipynb [hunk 1]
```diff
@@ -0,0 +1,299 @@
+{
```
Comment (yuzisun) [inline python/sklearnserver/sklearnserver/sklearn-multi-datatype-model.ipynb]: This should be moved to sklearn v1 examples

File: python/sklearnserver/custom_transformer/custom_transformer.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+from sklearn.base import TransformerMixin
+
+
+class DictToDFTransformer(TransformerMixin):
```
Comment (yuzisun) [inline python/sklearnserver/custom_transformer/custom_transformer.py:19]: This file is only used for sklearn pipeline example right?

Comment (Suresh-Nakkeran) [reply python/sklearnserver/custom_transformer/custom_transformer.py:19]: yes @yuzisun . It is used only for sklearnserver model test.

File: python/sklearnserver/sklearnserver/sklearn-multi-datatype-model.ipynb (hunk 1 above)
Comment (Suresh-Nakkeran) [reply python/sklearnserver/sklearnserver/sklearn-multi-datatype-model.ipynb]: I moved.

File: docs/samples/v1beta1/sklearn/v1/sklearn-multi-datatype-model/sklearn-multi-datatype-model.ipynb [hunk 3]
```diff
@@ -0,0 +1,299 @@
+{
```
Comment (yuzisun) [inline docs/samples/v1beta1/sklearn/v1/sklearn-multi-datatype-model/sklearn-multi-datatype-model.ipynb]: @Suresh-Nakkeran I have uploaded the model to `gs://kfserving-examples/models/sklearn/1.0/mixedtype/model.joblib`, can you help remove the above `model.joblib` file(keep the repo small) and add an inferenceservice yaml example file instead ?

File: docs/samples/v1beta1/sklearn/v1/sklearn-mixedtype-model/sklearn-mixedtype.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: "serving.kserve.io/v1beta1"
+kind: "InferenceService"
+metadata:
+  name: "sklearn-mixedtype"
+spec:
+  predictor:
+    sklearn:
+      storageUri: "gs://kfserving-examples/models/sklearn/1.0/mixedtype"
+      image: "suresh1233/sklearnserver:latest"
```
Comment (yuzisun) [inline docs/samples/v1beta1/sklearn/v1/sklearn-mixedtype-model/sklearn-mixedtype.yaml]:
```suggestion
      image: "kserve/sklearnserver:latest"
```

Comment (yuzisun) [reply docs/samples/v1beta1/sklearn/v1/sklearn-mixedtype-model/sklearn-mixedtype.yaml]: or maybe can simply remove the image field as the image is going to be published when 0.8 is released

Comment (Suresh-Nakkeran) [reply docs/samples/v1beta1/sklearn/v1/sklearn-mixedtype-model/sklearn-mixedtype.yaml]: sure @yuzisun

Comment (Suresh-Nakkeran) [reply docs/samples/v1beta1/sklearn/v1/sklearn-mixedtype-model/sklearn-mixedtype.yaml]: updated!

**Comment by yuzisun** [comment]
Thanks @Suresh-Nakkeran !

/lgtm
/approve

**Pull request merged**